components/browser/state/src/main/java/mozilla/components/browser/state/state/TabSessionState.kt

```diff
@@ -16,32 +16,37 @@ import java.util.UUID
  * parent. The parent tab is usually the tab that initiated opening this
  * tab (e.g. the user clicked a link with target="_blank" or selected
  * "open in new tab" or a "window.open" was triggered).
- * @property extensionState a map of web extension ids and extensions, that contains the overridden
- * values for this tab.
+ * @property extensionState a map of web extension ids to extensions,
+ * that contains the overridden values for this tab.
+ * @property readerState the [ReaderState] of this tab.
  */
 data class TabSessionState(
     override val id: String = UUID.randomUUID().toString(),
     override val content: ContentState,
     override val trackingProtection: TrackingProtectionState = TrackingProtectionState(),
     override val engineState: EngineState = EngineState(),
     val parentId: String? = null,
-    override val extensionState: Map<String, WebExtensionState> = emptyMap()
+    override val extensionState: Map<String, WebExtensionState> = emptyMap(),
+    val readerState: ReaderState = ReaderState()
```

Comment:
Small nit: I wonder if we should create some kind of `feature` sub-state that contains optional states that our features may add?

Reply:
Yes, was wondering about too, as we talked about this before, but wasn't sure. Let's punt on that and file a follow-up for grouping features together: https://github.com/mozilla-mobile/android-components/issues/5147

I will add a comment here as well.